core/src/main/java/org/apache/accumulo/core/client/impl/Tables.java

```diff
@@ -132,22 +111,35 @@ public static String getTableName(Instance instance, String tableId)
 
   /**
    * Get the TableMap from the cache. A new one will be populated when needed. Cache is cleared
-   * manually by calling {@link #clearCache(Instance)} or automatically cleared by ZooCache watcher
-   * created in {@link #getZooCache(Instance)}. See ACCUMULO-4778.
+   * manually by calling {@link #clearCache(Instance)}
    */
   private static TableMap getTableMap(final Instance instance) {
+
     TableMap map;
+
+    final ZooCache zc = getZooCache(instance);
+
+    map = getTableMap(instance, zc);
+
+    if (!map.isCurrent(zc)) {
+      instanceToMapCache.invalidate(instance.getInstanceID());
+      map = getTableMap(instance, zc);
+    }
```

Comment:
Do we need to synchronize on the TableMap cache?  Or does guava handle that?

Reply:
Guava handles that